debezium-testing/debezium-testing-testcontainers/src/main/java/io/debezium/testing/testcontainers/DebeziumContainer.java

```diff
@@ -84,45 +125,137 @@ private void registerConnectorToDebezium(final String payload, final String full
+    protected static Response executeRequest(Request request) throws IOException {
+        return CLIENT.newCall(request).execute();
+    }
 
-        try (Response response = client.newCall(request).execute()) {
-            return response.isSuccessful();
+    protected static Response executeRequestSuccessful(Request request) throws IOException {
+        Response response = executeRequest(request);
+        String responseBodyContent = "{empty response body}";
+        if (!response.isSuccessful()) {
+            ResponseBody responseBody = response.body();
+            if (null != responseBody) {
+                responseBodyContent = responseBody.string();
+                responseBody.close();
+            }
+            throw new IOException("Unexpected response: " + response + " Response Body: " + responseBodyContent);
         }
+        return response;
     }
 
-    /**
-     * Returns the "/connectors" endpoint.
-     */
-    public String getConnectors() {
-        return getTarget() + "/connectors/";
+    public boolean connectorIsNotRegistered(String connectorName) throws IOException {
+        try (Response response = executeRequest(new Request.Builder().url(getConnectorURI(connectorName)).get().build())) {
+            boolean connectorIsNotRegistered = response.code() == 404;
+            response.close();
+            return connectorIsNotRegistered;
+        }
     }
 
-    /**
-     * Returns the "/connectors/<connector>" endpoint.
-     */
-    public String getConnector(String connectorName) {
-        return getConnectors() + connectorName;
+    protected void deleteDebeziumConnector(String connectorName) throws IOException {
+        executeRequestSuccessful(new Request.Builder().url(getConnectorURI(connectorName)).delete().build()).close();
     }
 
-    /**
-     * Returns the "/connectors/<connector>/status" endpoint.
-     */
-    public String getConnectorStatus(String connectorName) {
-        return getConnectors() + connectorName + "/status";
+    public void deleteConnector(String connectorName) throws IOException {
+        deleteDebeziumConnector(connectorName);
+
+        Awaitility.await()
+                .atMost(10, TimeUnit.SECONDS)
+                .until(() -> connectorIsNotRegistered(connectorName));
     }
 
-    public String getTarget() {
-        return "http://" + getContainerIpAddress() + ":" + getMappedPort(8083);
+    public List<String> getRegisteredConnectors() throws IOException {
+        Request request = new Request.Builder().url(getConnectorsURI()).get().build();
+        try (ResponseBody responseBody = executeRequestSuccessful(request).body()) {
+            if (null != responseBody) {
+                String string = responseBody.string();
+                responseBody.close();
+                return MAPPER.readValue(string, new TypeReference<List<String>>() {
+                });
+            }
+        }
+        catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
The signature was old code, I removed the exception on the signature now. This is used for integration testing. When we are not able to talk to the Container it should imo be a RuntimeException and fail because the reason can only be a wrong state created by the testing code.